pymeasure/instruments/rigol/rigol_dp932u.py

```diff
@@ -0,0 +1,159 @@
+
+        :param adapter: The communication adapter (e.g., USB or GPIB) to connect to the instrument.
+        :type adapter: str
+
+        :param name: The name of the instrument.
+        :type name: str
+
+        :param kwargs: Additional arguments for instrument initialization.
+        :type kwargs: dict
+        """
+        super().__init__(adapter, name, **kwargs)
+
+    active_channel = Instrument.control(
+        ":INSTrument:NSELect?",
+        ":INSTrument:NSELect %d",
+        """Control the currently active channel (1, 2, or 3).
+
+        :param int value: Channel number to set as active.
+
+        :raises ValueError: If the channel number is outside the valid range [1, 3].
+        """,
+        validator=strict_discrete_set,
+        values=[1, 2, 3],
+    )
+
+    voltage = Instrument.control(
+        ":SOURce:VOLTage?",
+        ":SOURce:VOLTage %.3f",
+        """Control the voltage of the selected channel in Volts (0 to 32).
+
+        :param float value: Voltage level to set, within the range [0, 32].
+
+        :raises ValueError: If the voltage is outside the valid range.
+        """,
+        validator=strict_range,
+        values=[0, 32],
+    )
+
+    current = Instrument.control(
+        ":SOURce:CURRent?",
+        ":SOURce:CURRent %.3f",
+        """Control the current of the selected channel in Amps (0 to 3).
+
+        :param float value: Current level to set, within the range [0, 3].
+
+        :raises ValueError: If the current is outside the valid range.
+        """,
+        validator=strict_range,
+        values=[0, 3],
+    )
+
+    output_enabled = Instrument.control(
+        ":OUTPut:STATe?",
+        ":OUTPut:STATe %d",
+        """Control the output state of the selected channel as a boolean.
+
+        :param bool value: `True` to enable output, `False` to disable output.
+        """,
+        validator=strict_discrete_set,
+        values={False: 0, True: 1},
+        map_values=True,
+    )
+
+    connection_mode = Instrument.control(
+        ":OUTPut:PAIR?",
+        ":OUTPut:PAIR %s",
+        """Control the connection mode (OFF, PAR, or SER).
+
+        :param str value: Connection mode, either "OFF", "PAR" (parallel), or "SER" (series).
+        """,
+        validator=strict_discrete_set,
+        values={"OFF": "OFF", "PAR": "PAR", "SER": "SER", "PARALLEL": "PAR", "SERIES": "SER"},
+        map_values=True,
+    )
+
+    measure_voltage = Instrument.measurement(
+        ":MEASure:VOLTage:DC?",
+        """Measure the voltage of the currently selected channel in Volts.
+
+        :return: Measured voltage in Volts (float).
+        """
+    )
+
+    measure_current = Instrument.measurement(
+        ":MEASure:CURRent:DC?",
+        """Measure the current of the currently selected channel in Amps.
+
+        :return: Measured current in Amps (float).
+        """
+    )
+
+    def reset(self):
+        """
+        Resets the device to its factory defaults.
```

Comment:
Docstrings are in imperative mode:
```suggestion
        Reset the device to its factory defaults.
```